mobile/src/components/course/course.tsx

```diff
@@ -22,19 +24,44 @@ import {
 } from '~/hooks/hooks';
 import { courseModulesActions } from '~/store/actions';
 
+import { Category } from './components/components';
 import { CourseModules } from './components/course-modules/course-modules';
 import { styles } from './styles';
 
 const Course: FC = () => {
   const navigation = useAppNavigate();
   const { width } = useWindowDimensions();
   const dispatch = useAppDispatch();
-  const { course, dataStatus } = useAppSelector((state) => state.courses);
-  const dataCourse = course as CourseGetResponseDto;
+
+  const { user, course, dataStatus } = useAppSelector(({ auth, courses }) => ({
+    user: auth.user,
+    course: courses.course,
+    dataStatus: courses.dataStatus,
+  }));
+
+  const currentCategory = (course as CourseGetResponseDto).category;
```

Comment:
Can't we have situation on this screen when the course value is null, for example when we loading course data?